.emacs.d/lisp/behavior/pane-management.el

```diff
@@ -11,15 +11,12 @@ or if passed an argument, but just current frame if otherwise."
   (condition-case nil
       (let ((frame-list (frame-list)))
         (if (or (= (length frame-list) 1) arg)
-            (if (not (boundp 'server-process))
-                (if (y-or-n-p "Kill Emacs?")
-                    (kill-emacs)
-                  (message "Didn't do anything, promise!"))
-              (if (y-or-n-p "Kill Emacs?")
-                  (kill-emacs)
-                (if (y-or-n-p "Delete current frame?")
-                    (delete-frame)
-                  (message "Didn't do anything, promise!"))))
+            (if (y-or-n-p "Kill Emacs?")
```

Comment:
Let's explain to the user that we will also save buffers before killing emacs.  (`"Save buffers and kill Emacs?"` would work fine.)